test/e2e/network_test.go

```diff
@@ -148,6 +152,54 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// collectDeploymentLogs collects logs of a deployment
+// to the Artifacts directory
+func (tc *testContext) collectDeploymentLogs(deployment *appsv1.Deployment) {
+	// map of labels expected to be on each pod in the deployment
+	matchLabels := deployment.Spec.Selector.MatchLabels
+	if len(matchLabels) == 0 {
+		log.Printf("matchLabels is empty")
+		return
+	}
+	var keyValPairs []string
+	for key, value := range matchLabels {
+		keyValPairs = append(keyValPairs, key+"="+value)
+	}
+	labelSelector := strings.Join(keyValPairs, ",")
+	tc.writePodLogs(labelSelector)
+}
+
+// getLogs uses a podLabelSelector and returns the logs
+// associated with that pod
```

Comment:
Could we make this function's comment a bit more clear? Maybe `podLabelSelector` --> `label selector` and `that pod` --> `that job`?

Reply:
Also no need for it to be a 2-line comment

Reply:
> Could we make this function's comment a bit more clear? Maybe `podLabelSelector` --> `label selector` and `that pod` --> `that job`?

We are also collecting logs from pods from deployments, so `that pod` is correct here.

Reply:
My thought was that one call to `getLogs` can collect logs from multiple pods, so perhaps `those pods`?

Reply:
What about `those pod(s)` or `with each pod`?

Reply:
`each pod` sounds good